EIPS/eip-5805.md

```diff
@@ -0,0 +1,401 @@
+
+When a "delegator" delegates its tokens voting power to a "delegatee", its balance is added to the voting power of the delegatee. If the delegator changes its delegation, the voting power is subtracted from the old delegatee's voting power and added to the new delegate's voting power. The voting power of each account is tracked through time so that it is possible to query its value in the past. With tokens being delegated to at most one delegate at a given point in time, double voting is prevented.
+
+Whenever tokens are transferred from one account to another, the associated voting power should be deducted from the sender's delegate and added to the receiver's delegate.
+
+Tokens that are delegated to `address(0)` should not be tracked. This allows users to optimize the gas cost of their token transfers by skipping the checkpoint update for their delegate.
+
+To accommodate different types of chains, we want the voting checkpoint system to support different forms of time tracking. On the Ethereum mainnet, using block numbers provides backward compatibility with applications that historically use it. On the other hand, using timestamps provides better semantics for end users, and accommodates use cases where the duration is expressed in seconds. Other monotonic functions could also be deemed relevant by developers based on the characteristics of future applications and blockchains.
+
+Both timestamps, block numbers, and other possible modes use the same external interfaces. This allows transparent binding of third-party contracts, such as governor systems, to the vote tracking built into the voting contracts. For this to be effective, the voting contracts must, in addition to all the vote-tracking functions, expose the current value used for time-tracking.
+
+### Methods
+
+#### now
+
+This function returns the current timepoint. It could be `block.timestamp`, `block.number` (or any other **non-decreasing** function) depending on the mode the contract is operating on.
+
+- If operating using **block number**, then this function SHOULD be implemented.
+- If operating using **timestamp**, then this function MUST be implemented.
+- If operating using any other mode, then this function MUST be implemented.
+
+This function is thus optional, and its absence should be considered as a marker of the contract operating using block number. (This makes this EIP compatible with pre-existing voting contracts)
+
+```yaml
+- name: now
+  type: function
+  stateMutability: view
+  inputs: []
+  outputs:
+    - name: timepoint
+      type: uint256
+```
+
+#### getVotes
+
+This function returns the current voting weight of an account. This corresponds to all the voting power delegated to it at the moment this function is called.
+
+As tokens delegated to `address(0)` should not be counted/snapshoted, `getVotes(0)` SHOULD always return `0`.
```

Comment:
My intention is to say that if an implementer wants to track tokens delegated to `address(0)`, and return this non 0 value in `getVotes(0)` and `getPastVotes(0, x)` ... it would not break the key mechanism of this EIP.

If devs what to track address(0) ... that is ok ... but users should not assume that it will be the case.